cmake/sca/codechecker/sca.cmake

```diff
@@ -0,0 +1,47 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2023, Basalte bv
+
+find_program(CODECHECKER_EXE CodeChecker REQUIRED)
+message(STATUS "Found CodeChecker: ${CODECHECKER_EXE}")
+
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+if(CODECHECKER_CONFIG)
```

Comment:
I think that users would likely want to configure this in any case, so imposing other default rules would've been confusing IMO.

Sure, putting the checker out-of-tree is as simple as putting the `sca.cmake` file in a registered `sca_root`.
The native SCA support in zephyr is currently only limited to `sparse` (also with the tool's defaults?), so I thought to share this as the setup is really easy, and the benefits are great. 
And hopefully it would improve the `zephyr`'s codebase as well by more users running SCA.